src/core/dynamic_batch_scheduler.cc

```diff
@@ -506,19 +544,31 @@ DynamicBatchScheduler::GetDynamicBatch()
   return wait_ns / 1000;
 }
 
+// TODO: Replace cache_hit flag with struct of cache info
+//       or use cache manager to track logic
 void
 DynamicBatchScheduler::DelegateResponse(
-    std::unique_ptr<InferenceRequest>& request)
+    std::unique_ptr<InferenceRequest>& request, uint64_t request_hash, bool cache_hit)
```

Comment:
This can probably be cleaned up by implementing a ResponseCacheManager to keep track of cache state rather than passing flags to `DelegateResponse`